packages/loader/container-loader/src/container.ts

```diff
@@ -1522,11 +1535,18 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         deltaManager.inboundSignal.pause();
 
         deltaManager.on("connect", (details: IConnectionDetails, opsBehind?: number) => {
-            // Back-compat for new client and old server.
-            this._audience.clear();
+            // Store the initial clients so that they can be submitted to the
+            // protocol handler when it is created.
+            this._initialClients = details.initialClients ?? [];
```

Comment:
I'd probably keep this._initialClients undefined if we have protocol - no need to keep this memory around. We do support 500 connections per doc, so it could be couple Kbs of memory that is not needed.
Somewhat similar feedback RE getting rid of this memory after creating audience

Reply:
good call, thank you